diff --git a/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC.hs b/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC.hs
index 31fa503a..8766276d 100644
--- a/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC.hs
+++ b/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC.hs
@@ -30,7 +30,6 @@ import Distribution.Simple.Utils
 import qualified Distribution.ModuleName as ModuleName
 import Distribution.Simple.Program
 import qualified Distribution.Simple.Program.Ar    as Ar
-import qualified Distribution.Simple.Program.Ld    as Ld
 import Distribution.Simple.Program.GHC
 import Distribution.Simple.Setup
 import qualified Distribution.Simple.Setup as Cabal
@@ -359,7 +358,7 @@ buildOrReplLib forRepl verbosity numJobs pkg_descr lbi lib clbi = do

       whenGHCiLib $ do
         (ldProg, _) <- requireProgram verbosity ldProgram (withPrograms lbi)
-        Ld.combineObjectFiles verbosity ldProg
+        Internal.combineObjectFiles verbosity lbi ldProg
           ghciLibFilePath ghciObjFiles

       whenSharedLib False $
diff --git a/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC/Internal.hs b/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC/Internal.hs
index e7e2f953..47a879af 100644
--- a/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC/Internal.hs
+++ b/Data/Array/Accelerate/LLVM/Native/Distribution/Simple/GHC/Internal.hs
@@ -19,6 +19,7 @@ module Data.Array.Accelerate.LLVM.Native.Distribution.Simple.GHC.Internal (
   getHaskellObjects,
   mkGhcOptPackages,
   profDetailLevelFlag,
+  combineObjectFiles

 ) where

@@ -28,9 +29,12 @@ import Distribution.Backpack
 import Distribution.PackageDescription as PD hiding (Flag)
 import Distribution.Simple.Compiler hiding (Flag)
 import Distribution.Simple.LocalBuildInfo
+import Distribution.Simple.Program (ConfiguredProgram)
 import Distribution.Simple.Program.GHC
+import qualified Distribution.Simple.Program.Ld as Ld
 import Distribution.Simple.Setup
 import Distribution.Simple
+import Distribution.Verbosity (Verbosity)
 import qualified Distribution.ModuleName as ModuleName

 import qualified Data.Map as Map
@@ -113,3 +117,10 @@ allLibModules :: Library -> ComponentLocalBuildInfo -> [ModuleName.ModuleName]
 allLibModules lib _ = libModules lib
 #endif

+combineObjectFiles :: Verbosity -> LocalBuildInfo -> ConfiguredProgram
+                   -> FilePath -> [FilePath] -> IO ()
+#if MIN_VERSION_Cabal(2,1,0)
+combineObjectFiles = Ld.combineObjectFiles
+#else
+combineObjectFiles v _ = Ld.combineObjectFiles v
+#endif
