commit 194eaa0d3dbafd593eeb12deb57374476a1adf4f
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Fri Mar 15 18:27:52 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/Data/Aeson/Types/Internal.hs b/Data/Aeson/Types/Internal.hs
index abe5afc..319225c 100644
--- a/Data/Aeson/Types/Internal.hs
+++ b/Data/Aeson/Types/Internal.hs
@@ -157,8 +157,10 @@ instance Monad IResult where
     IError path err >>= _ = IError path err
     {-# INLINE (>>=) #-}
 
+#if !(MIN_VERSION_base(4,13,0))
     fail = Fail.fail
     {-# INLINE fail #-}
+#endif
 
 instance Fail.MonadFail IResult where
     fail err = IError [] err
@@ -172,8 +174,10 @@ instance Monad Result where
     Error err >>= _ = Error err
     {-# INLINE (>>=) #-}
 
+#if !(MIN_VERSION_base(4,13,0))
     fail = Fail.fail
     {-# INLINE fail #-}
+#endif
 
 instance Fail.MonadFail Result where
     fail err = Error err
@@ -287,8 +291,10 @@ instance Monad Parser where
     {-# INLINE (>>=) #-}
     return = pure
     {-# INLINE return #-}
+#if !(MIN_VERSION_base(4,13,0))
     fail = Fail.fail
     {-# INLINE fail #-}
+#endif
 
 instance Fail.MonadFail Parser where
     fail msg = Parser $ \path kf _ks -> kf (reverse path) msg
