diff -ru blank-canvas-0.6.3.orig/Graphics/Blank/Style.hs blank-canvas-0.6.3/Graphics/Blank/Style.hs
--- blank-canvas-0.6.3.orig/Graphics/Blank/Style.hs	2018-04-07 20:04:46.000000000 -0400
+++ blank-canvas-0.6.3/Graphics/Blank/Style.hs	2019-03-15 18:46:01.470815640 -0400
@@ -189,6 +189,8 @@
     , rebeccapurple
     ) where
 
+import qualified Control.Monad.Fail as Fail
+
 import qualified Data.Colour as Colour
 import           Data.Colour hiding (black, transparent)
 import qualified Data.Colour.Names as Names
@@ -248,7 +250,7 @@
 -- |
 -- Takes a string naming a 'Colour' (must be all lowercase) and returns it. Fails if
 -- the name is not recognized.
-readColourName :: Monad m => String -> m (Colour Double)
+readColourName :: Fail.MonadFail m => String -> m (Colour Double)
 readColourName "rebeccapurple" = return rebeccapurple
 readColourName name            = Names.readColourName name
 
