diff -ru blaze-builder-0.4.0.2.orig/Blaze/ByteString/Builder/Internal/Write.hs blaze-builder-0.4.0.2/Blaze/ByteString/Builder/Internal/Write.hs
--- blaze-builder-0.4.0.2.orig/Blaze/ByteString/Builder/Internal/Write.hs	2016-04-18 21:47:02.000000000 +0200
+++ blaze-builder-0.4.0.2/Blaze/ByteString/Builder/Internal/Write.hs	2017-09-14 22:57:46.824329914 +0200
@@ -56,6 +56,9 @@
 #if !MIN_VERSION_base(4,8,0)
 import Data.Monoid
 #endif
+#if !MIN_VERSION_base(4,11,0)
+import Data.Semigroup
+#endif
 
 ------------------------------------------------------------------------------
 -- Poking a buffer and writing to a buffer
@@ -119,23 +122,31 @@
     getBound $ write $ error $
     "getBound' called from " ++ msg ++ ": write bound is not data-independent."
 
+instance Semigroup Poke where
+  {-# INLINE (<>) #-}
+  (Poke po1) <> (Poke po2) = Poke $ po1 >=> po2
+
 instance Monoid Poke where
   {-# INLINE mempty #-}
   mempty = Poke $ return
 
   {-# INLINE mappend #-}
-  (Poke po1) `mappend` (Poke po2) = Poke $ po1 >=> po2
+  mappend = (<>)
 
   {-# INLINE mconcat #-}
   mconcat = foldr mappend mempty
 
+instance Semigroup Write where
+  {-# INLINE (<>) #-}
+  (Write bound1 w1) <> (Write bound2 w2) =
+    Write (bound1 + bound2) (w1 `mappend` w2)
+
 instance Monoid Write where
   {-# INLINE mempty #-}
   mempty = Write 0 mempty
 
   {-# INLINE mappend #-}
-  (Write bound1 w1) `mappend` (Write bound2 w2) =
-    Write (bound1 + bound2) (w1 `mappend` w2)
+  mappend = (<>)
 
   {-# INLINE mconcat #-}
   mconcat = foldr mappend mempty
Only in blaze-builder-0.4.0.2/Blaze/ByteString/Builder/Internal: Write.hs~
