diff -ru boxes-0.1.4.orig/Text/PrettyPrint/Boxes.hs boxes-0.1.4/Text/PrettyPrint/Boxes.hs
--- boxes-0.1.4.orig/Text/PrettyPrint/Boxes.hs	2015-01-14 11:19:44.000000000 -0500
+++ boxes-0.1.4/Text/PrettyPrint/Boxes.hs	2018-01-23 19:44:46.961988690 -0500
@@ -97,7 +97,9 @@
 
     ) where
 
-#if MIN_VERSION_base(4,8,0)
+#if MIN_VERSION_base(4,11,0)
+import Prelude hiding ( (<>), Word )
+#elif MIN_VERSION_base(4,8,0)
 import Prelude hiding (Word)
 #endif
 
diff -ru boxes-0.1.4.orig/Text/PrettyPrint/Tests.hs boxes-0.1.4/Text/PrettyPrint/Tests.hs
--- boxes-0.1.4.orig/Text/PrettyPrint/Tests.hs	2015-01-14 11:19:44.000000000 -0500
+++ boxes-0.1.4/Text/PrettyPrint/Tests.hs	2018-01-23 19:45:09.913989268 -0500
@@ -4,6 +4,10 @@
 import Control.Monad
 import System.Exit (exitFailure, exitSuccess)
 
+#if MIN_VERSION_base(4,11,0)
+import Prelude hiding ((<>))
+#endif
+
 instance Arbitrary Alignment where
   arbitrary = elements [ AlignFirst
                        , AlignCenter1
