commit 79490be79c4ad2e07fc8e43a9887fc21f1907f76
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Wed Jun 12 18:52:03 2019 -0400

    Allow building with Cabal-3.0.*

diff --git a/src/Distribution/Extra/Doctest.hs b/src/Distribution/Extra/Doctest.hs
index 1beb9d2..94ca62b 100644
--- a/src/Distribution/Extra/Doctest.hs
+++ b/src/Distribution/Extra/Doctest.hs
@@ -69,8 +69,10 @@ import Distribution.PackageDescription
        PackageDescription (), TestSuite (..))
 import Distribution.Simple
        (UserHooks (..), autoconfUserHooks, defaultMainWithHooks, simpleUserHooks)
+#if !(MIN_VERSION_Cabal(3,0,0))
 import Distribution.Simple.BuildPaths
        (autogenModulesDir)
+#endif
 import Distribution.Simple.Compiler
        (PackageDB (..), showCompilerId)
 import Distribution.Simple.LocalBuildInfo
@@ -79,7 +81,11 @@ import Distribution.Simple.LocalBuildInfo
 import Distribution.Simple.Setup
        (BuildFlags (buildDistPref, buildVerbosity), HaddockFlags (haddockDistPref, haddockVerbosity), fromFlag, emptyBuildFlags)
 import Distribution.Simple.Utils
-       (createDirectoryIfMissingVerbose, findFile, rewriteFile)
+       (createDirectoryIfMissingVerbose, findFile)
+#if !(MIN_VERSION_Cabal(3,0,0))
+import Distribution.Simple.Utils
+       (rewriteFile)
+#endif
 import Distribution.Text
        (display, simpleParse)
 import System.FilePath
@@ -103,6 +109,9 @@ import Distribution.Types.GenericPackageDescription
 import Distribution.PackageDescription
        (CondTree (..))
 #endif
+#if MIN_VERSION_Cabal(2,5,0)
+import Distribution.Types.LibraryName (libraryNameString)
+#endif
 
 #if MIN_VERSION_directory(1,2,2)
 import System.Directory
@@ -432,7 +441,9 @@ generateBuildModule testSuiteName flags pkg lbi = do
        isSpecific _                     = False
 
     mbLibraryName :: Library -> Name
-#if MIN_VERSION_Cabal(2,0,0)
+#if MIN_VERSION_Cabal(2,5,0)
+    mbLibraryName = NameLib . fmap unUnqualComponentName . libraryNameString . libName
+#elif MIN_VERSION_Cabal(2,0,0)
     -- Cabal-2.0 introduced internal libraries, which are named.
     mbLibraryName = NameLib . fmap unUnqualComponentName . libName
 #else
