diff -ru colour-2.3.4.orig/Data/Colour/Names.hs colour-2.3.4/Data/Colour/Names.hs
--- colour-2.3.4.orig/Data/Colour/Names.hs	2017-12-03 22:43:01.000000000 -0500
+++ colour-2.3.4/Data/Colour/Names.hs	2019-03-15 09:50:01.875413167 -0400
@@ -182,10 +182,11 @@
 where
 
 import Prelude hiding (tan)
+import qualified Control.Monad.Fail as Fail
 import Data.Colour.SRGB
 import Data.Colour (black)
 
-readColourName :: (Monad m, Ord a, Floating a) => String -> m (Colour a)
+readColourName :: (Fail.MonadFail m, Ord a, Floating a) => String -> m (Colour a)
 readColourName "aliceblue" = return aliceblue
 readColourName "antiquewhite" = return antiquewhite
 readColourName "aqua" = return aqua
