commit e03f3dfcffdda101ff48b508a80cc5d99cba89e2
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Thu Feb 21 06:44:43 2019 -0500

    Fix the build with GHC 8.9

diff --git a/src/Data/Constraint/Forall.hs b/src/Data/Constraint/Forall.hs
index 3932f29..185fc95 100644
--- a/src/Data/Constraint/Forall.hs
+++ b/src/Data/Constraint/Forall.hs
@@ -142,8 +142,14 @@ instance Forall (R p t a) => Q p t a
 class Forall (Q p t) => ForallT (p :: k4 -> Constraint) (t :: (k1 -> k2) -> k3 -> k4)
 instance Forall (Q p t) => ForallT p t
 
+#if __GLASGOW_HASKELL__ >= 809
+# define KVS(kvs) kvs
+#else
+# define KVS(kvs)
+#endif
+
 -- | Instantiate a quantified @'ForallT' p t@ constraint at types @f@ and @a@.
-instT :: forall (p :: k4 -> Constraint) (t :: (k1 -> k2) -> k3 -> k4) (f :: k1 -> k2) (a :: k3). ForallT p t :- p (t f a)
+instT :: forall KVS(k4 k1 k2 k3) (p :: k4 -> Constraint) (t :: (k1 -> k2) -> k3 -> k4) (f :: k1 -> k2) (a :: k3). ForallT p t :- p (t f a)
 instT = Sub $
   case inst :: Forall (Q p t) :- Q p t f of { Sub Dict ->
   case inst :: Forall (R p t f) :- R p t f a of
