diff -ru criterion-1.2.2.0.orig/Criterion/Types.hs criterion-1.2.2.0/Criterion/Types.hs
--- criterion-1.2.2.0.orig/Criterion/Types.hs	2017-08-05 18:35:19.000000000 +0200
+++ criterion-1.2.2.0/Criterion/Types.hs	2017-09-15 11:45:03.538012575 +0200
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE Trustworthy #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE DeriveDataTypeable, DeriveGeneric, GADTs, RecordWildCards #-}
@@ -74,6 +75,7 @@
 -- Temporary: to support pre-AMP GHC 7.8.4:
 import Control.Applicative
 import Data.Monoid
+import Data.Semigroup
 
 import Control.DeepSeq (NFData(rnf))
 import Control.Exception (evaluate)
@@ -650,9 +652,14 @@
             _ -> fail $ "get for OutlierEffect: unexpected " ++ show i
 instance NFData OutlierEffect
 
+instance Semigroup Outliers where
+    (<>) = addOutliers
+
 instance Monoid Outliers where
     mempty  = Outliers 0 0 0 0 0
+#if !(MIN_VERSION_base(4,11,0))
     mappend = addOutliers
+#endif
 
 addOutliers :: Outliers -> Outliers -> Outliers
 addOutliers (Outliers s a b c d) (Outliers t w x y z) =
Only in criterion-1.2.2.0/Criterion: Types.hs.orig
diff -ru criterion-1.2.2.0.orig/criterion.cabal criterion-1.2.2.0/criterion.cabal
--- criterion-1.2.2.0.orig/criterion.cabal	2017-08-05 18:35:19.000000000 +0200
+++ criterion-1.2.2.0/criterion.cabal	2017-09-15 11:45:03.542012553 +0200
@@ -101,6 +101,7 @@
     mwc-random >= 0.8.0.3,
     optparse-applicative >= 0.13,
     parsec >= 3.1.0,
+    semigroups,
     statistics >= 0.14 && < 0.15,
     text >= 0.11,
     time,
