commit dd847e8d29133f5a5c420e728023f92e2023c283
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Thu Dec 28 11:21:03 2017 -0500

    Remove deprecated GCStats-based interface

    Fixes #171.

diff --git a/Criterion/Measurement.hs b/Criterion/Measurement.hs
index d352bf6..d006933 100644
--- a/Criterion/Measurement.hs
+++ b/Criterion/Measurement.hs
@@ -5,12 +5,6 @@
 {-# LANGUAGE BangPatterns, CPP, ForeignFunctionInterface,
     ScopedTypeVariables #-}

-#if MIN_VERSION_base(4,10,0)
--- Disable deprecation warnings for now until we remove the use of getGCStats
--- and applyGCStats for good
-{-# OPTIONS_GHC -Wno-deprecations #-}
-#endif
-
 -- |
 -- Module      : Criterion.Measurement
 -- Copyright   : (c) 2009-2014 Bryan O'Sullivan
@@ -38,9 +32,6 @@ module Criterion.Measurement
     , measured
     , applyGCStatistics
     , threshold
-      -- * Deprecated
-    , getGCStats
-    , applyGCStats
     ) where

 import Criterion.Types (Benchmarkable(..), Measured(..))
@@ -52,9 +43,10 @@ import Data.Int (Int64)
 import Data.List (unfoldr)
 import Data.Word (Word64)
 import GHC.Generics (Generic)
-import GHC.Stats (GCStats(..))
 #if MIN_VERSION_base(4,10,0)
 import GHC.Stats (RTSStats(..), GCDetails(..))
+#else
+import GHC.Stats (GCStats(..))
 #endif
 import System.Mem (performGC)
 import Text.Printf (printf)
@@ -66,9 +58,9 @@ import qualified GHC.Stats as Stats
 -- 'gcStatsCurrentBytesUsed' and 'gcStatsCurrentBytesSlop' all are cumulative values since
 -- the program started.
 --
--- 'GCStatistics' is cargo-culted from the 'GCStats' data type that "GHC.Stats"
--- has. Since 'GCStats' was marked as deprecated and will be removed in GHC 8.4,
--- we use 'GCStatistics' to provide a backwards-compatible view of GC statistics.
+-- 'GCStatistics' is cargo-culted from the @GCStats@ data type that "GHC.Stats"
+-- used to export. Since @GCStats@ was removed in GHC 8.4, @criterion@ uses
+-- 'GCStatistics' to provide a backwards-compatible view of GC statistics.
 data GCStatistics = GCStatistics
     { -- | Total number of bytes allocated
     gcStatsBytesAllocated :: !Int64
@@ -115,18 +107,6 @@ data GCStatistics = GCStatistics
 -- | Try to get GC statistics, bearing in mind that the GHC runtime
 -- will throw an exception if statistics collection was not enabled
 -- using \"@+RTS -T@\".
-{-# DEPRECATED getGCStats
-      ["GCStats has been deprecated in GHC 8.2. As a consequence,",
-       "getGCStats has also been deprecated in favor of getGCStatistics.",
-       "getGCStats will be removed in the next major criterion release."] #-}
-getGCStats :: IO (Maybe GCStats)
-getGCStats =
-  (Just `fmap` Stats.getGCStats) `Exc.catch` \(_::Exc.SomeException) ->
-  return Nothing
-
--- | Try to get GC statistics, bearing in mind that the GHC runtime
--- will throw an exception if statistics collection was not enabled
--- using \"@+RTS -T@\".
 getGCStatistics :: IO (Maybe GCStatistics)
 #if MIN_VERSION_base(4,10,0)
 -- Use RTSStats/GCDetails to gather GC stats
@@ -331,30 +311,6 @@ measured = Measured {

 -- | Apply the difference between two sets of GC statistics to a
 -- measurement.
-{-# DEPRECATED applyGCStats
-      ["GCStats has been deprecated in GHC 8.2. As a consequence,",
-       "applyGCStats has also been deprecated in favor of applyGCStatistics.",
-       "applyGCStats will be removed in the next major criterion release."] #-}
-applyGCStats :: Maybe GCStats
-             -- ^ Statistics gathered at the __end__ of a run.
-             -> Maybe GCStats
-             -- ^ Statistics gathered at the __beginning__ of a run.
-             -> Measured
-             -- ^ Value to \"modify\".
-             -> Measured
-applyGCStats (Just end) (Just start) m = m {
-    measAllocated          = diff bytesAllocated
-  , measNumGcs             = diff numGcs
-  , measBytesCopied        = diff bytesCopied
-  , measMutatorWallSeconds = diff mutatorWallSeconds
-  , measMutatorCpuSeconds  = diff mutatorCpuSeconds
-  , measGcWallSeconds      = diff gcWallSeconds
-  , measGcCpuSeconds       = diff gcCpuSeconds
-  } where diff f = f end - f start
-applyGCStats _ _ m = m
-
--- | Apply the difference between two sets of GC statistics to a
--- measurement.
 applyGCStatistics :: Maybe GCStatistics
                   -- ^ Statistics gathered at the __end__ of a run.
                   -> Maybe GCStatistics
