diff -ru darcs-2.14.1.orig/src/Darcs/Util/Compat.hs darcs-2.14.1/src/Darcs/Util/Compat.hs
--- darcs-2.14.1.orig/src/Darcs/Util/Compat.hs	2018-06-24 14:17:58.000000000 -0400
+++ darcs-2.14.1/src/Darcs/Util/Compat.hs	2018-07-31 09:16:23.829958494 -0400
@@ -36,6 +36,9 @@
 import System.Posix.Files ( stdFileMode )
 import System.Posix.IO ( openFd, closeFd, stdOutput, stdError,
                          dupTo, defaultFileFlags, exclusive,
+#if MIN_VERSION_unix(2,8,0)
+                         creat,
+#endif
                          OpenMode(WriteOnly) )
 import System.Posix.Types ( Fd(..) )
 
@@ -121,7 +124,12 @@
 
 sloppyAtomicCreate :: FilePath -> IO ()
 sloppyAtomicCreate fp
-    = do fd <- openFd fp WriteOnly (Just stdFileMode) flags
+    = do fd <- openFd fp WriteOnly
+#if MIN_VERSION_unix(2,8,0)
+                      flags {creat = Just stdFileMode}
+#else
+                      (Just stdFileMode) flags
+#endif
          closeFd fd
   where flags = defaultFileFlags { exclusive = True }
 
