commit 01c6185d317d4171c08908074ab01d83fc694ceb
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Mon Feb 26 12:40:06 2018 -0500

    Add a Semigroup instance for Following

diff --git a/src/Diagrams/TwoD/Path/Follow.hs b/src/Diagrams/TwoD/Path/Follow.hs
index 585913b..412acdf 100644
--- a/src/Diagrams/TwoD/Path/Follow.hs
+++ b/src/Diagrams/TwoD/Path/Follow.hs
@@ -61,7 +61,7 @@ import           Data.Monoid.SemiDirectProduct.Strict
 --
 newtype Following n
   = Following { unFollowing :: Semi (Trail' Line V2 n) (Angle n) }
-  deriving (Monoid)
+  deriving (Monoid, Semigroup)
 
 -- | Note this is only an iso when considering trails equivalent up to
 --   rotation.
