commit 2f7655be01d0f983de3ad91f2a0ad936fa5c875d
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Thu Jan 25 10:54:41 2018 -0500

    Add a Semigroup (Render NullBackend v n) instance

diff --git a/src/Diagrams/Core/Types.hs b/src/Diagrams/Core/Types.hs
index 56b86ab..625eea2 100644
--- a/src/Diagrams/Core/Types.hs
+++ b/src/Diagrams/Core/Types.hs
@@ -966,9 +966,14 @@ data NullBackend
 --
 -- because it overlaps with the Renderable instance for NullPrim.
 
+instance Semigroup (Render NullBackend v n) where
+  _ <> _ = NullBackendRender
+
 instance Monoid (Render NullBackend v n) where
-  mempty      = NullBackendRender
-  mappend _ _ = NullBackendRender
+  mempty  = NullBackendRender
+#if !MIN_VERSION_base(4,11,0)
+  mappend = (<>)
+#endif
 
 instance Backend NullBackend v n where
   data Render NullBackend v n = NullBackendRender
