commit 7942c5ca3e28d2a546e5d351ded87f454655a58c
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Fri Mar 29 13:57:16 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/Data/DList.hs b/Data/DList.hs
index 06f1ce8..27a016e 100644
--- a/Data/DList.hs
+++ b/Data/DList.hs
@@ -65,6 +65,7 @@ import Prelude hiding (concat, foldr, map, head, tail, replicate)
 import qualified Data.List as List
 import Control.DeepSeq (NFData(..))
 import Control.Monad as M
+import qualified Control.Monad.Fail as Fail
 import Data.Function (on)
 import Data.String (IsString(..))
 
@@ -278,6 +279,12 @@ instance Monad DList where
   return   = pure
   {-# INLINE return #-}
 
+#if !(MIN_VERSION_base(4,13,0))
+  fail     = Fail.fail
+  {-# INLINE fail #-}
+#endif
+
+instance Fail.MonadFail DList where
   fail _   = empty
   {-# INLINE fail #-}
 
