commit 3f975944a59e21e3bbe4cb52f466bb8ef1a26801
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Fri Mar 15 08:38:04 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/src/Control/Monad/Catch/Pure.hs b/src/Control/Monad/Catch/Pure.hs
index 40f58ac..4e433b5 100644
--- a/src/Control/Monad/Catch/Pure.hs
+++ b/src/Control/Monad/Catch/Pure.hs
@@ -54,6 +54,7 @@ import Prelude hiding (catch, foldr)
 
 import Control.Applicative
 import Control.Monad.Catch
+import qualified Control.Monad.Fail as Fail
 import Control.Monad.Reader as Reader
 import Control.Monad.RWS
 #if __GLASGOW_HASKELL__ < 710
@@ -109,6 +110,11 @@ instance Monad m => Monad (CatchT m) where
   CatchT m >>= k = CatchT $ m >>= \ea -> case ea of
     Left e -> return (Left e)
     Right a -> runCatchT (k a)
+#if !(MIN_VERSION_base(4,13,0))
+  fail = Fail.fail
+#endif
+
+instance Monad m => Fail.MonadFail (CatchT m) where
   fail = CatchT . return . Left . toException . userError
 
 instance MonadFix m => MonadFix (CatchT m) where
