commit 3b85d9ead4a52812e67c928eef10fff64c8f5c93
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Sat Oct 27 22:01:34 2018 -0400

    Allow building with template-haskell-2.15.0.0

diff --git a/src/Control/Exhaustive.hs b/src/Control/Exhaustive.hs
index a77dd4c..facf4fe 100644
--- a/src/Control/Exhaustive.hs
+++ b/src/Control/Exhaustive.hs
@@ -179,9 +179,17 @@ constructors t =
          return (Just ctors)
        TyConI (NewtypeD _ _ _ _ ctor _) ->
          return (Just [ctor])
-       TyConI (DataInstD _ _ _ _ ctors _) ->
+       TyConI (DataInstD _ _
+# if MIN_VERSION_template_haskell(2,15,0)
+                           _
+# endif
+                           _ _ ctors _) ->
          return (Just ctors)
-       TyConI (NewtypeInstD _ _ _ _ ctor _) ->
+       TyConI (NewtypeInstD _ _
+# if MIN_VERSION_template_haskell(2,15,0)
+                              _
+# endif
+                              _ _ ctor _) ->
          return (Just [ctor])
        _ -> return Nothing
 #else
