diff --git a/System/FileLock/Internal/Flock.hsc b/System/FileLock/Internal/Flock.hsc
index 6a5d9ea..559217a 100644
--- a/System/FileLock/Internal/Flock.hsc
+++ b/System/FileLock/Internal/Flock.hsc
@@ -12,7 +12,11 @@ import Data.Bits
 import Foreign.C.Error
 import Foreign.C.Types
 import System.Posix.Files
-import System.Posix.IO (openFd, closeFd, defaultFileFlags, OpenMode(..))
+import System.Posix.IO ( openFd, closeFd, defaultFileFlags, OpenMode(..)
+#if MIN_VERSION_unix(2,8,0)
+                       , creat
+#endif
+                       )
 import System.Posix.Types
 import Prelude
 
@@ -38,7 +42,12 @@ unlock :: Lock -> IO ()
 unlock fd = closeFd fd
 
 open :: FilePath -> IO Fd
-open path = openFd path WriteOnly (Just stdFileMode) defaultFileFlags
+open path = openFd path WriteOnly
+#if MIN_VERSION_unix(2,8,0)
+                   (defaultFileFlags {creat = Just stdFileMode})
+#else
+                   (Just stdFileMode) defaultFileFlags
+#endif
 
 flock :: Fd -> Bool -> Bool -> IO Bool
 flock (Fd fd) exclusive block = do
