commit ef1cdd4ca6afd540ba142b545b9a5937ff8810f3
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Fri Mar 15 14:18:23 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/src/Control/Monad/Trans/Free.hs b/src/Control/Monad/Trans/Free.hs
index 555cc91..1d83846 100644
--- a/src/Control/Monad/Trans/Free.hs
+++ b/src/Control/Monad/Trans/Free.hs
@@ -308,9 +308,11 @@ instance (Functor f, Monad m) => Monad (FreeT f m) where
     Pure a -> runFreeT (f a)
     Free w -> return (Free (fmap (>>= f) w))
 
-  fail = Fail.fail
+#if !(MIN_VERSION_base(4,13,0))
+  fail e = FreeT (fail e)
+#endif
 
-instance (Functor f, Monad m) => Fail.MonadFail (FreeT f m) where
+instance (Functor f, Fail.MonadFail m) => Fail.MonadFail (FreeT f m) where
   fail e = FreeT (fail e)
 
 instance MonadTrans (FreeT f) where
diff --git a/src/Control/Monad/Trans/Free/Ap.hs b/src/Control/Monad/Trans/Free/Ap.hs
index f739d2b..6239025 100644
--- a/src/Control/Monad/Trans/Free/Ap.hs
+++ b/src/Control/Monad/Trans/Free/Ap.hs
@@ -307,9 +307,11 @@ instance (Applicative f, Applicative m, Monad m) => Monad (FreeT f m) where
   FreeT m >>= f = FreeT $ m >>= \v -> case v of
     Pure a -> runFreeT (f a)
     Free w -> return (Free (fmap (>>= f) w))
-  fail = Fail.fail
+#if !(MIN_VERSION_base(4,13,0))
+  fail e = FreeT (fail e)
+#endif
 
-instance (Applicative f, Applicative m, Monad m) => Fail.MonadFail (FreeT f m) where
+instance (Applicative f, Applicative m, Fail.MonadFail m) => Fail.MonadFail (FreeT f m) where
   fail e = FreeT (fail e)
 
 instance MonadTrans (FreeT f) where
diff --git a/src/Control/Monad/Trans/Iter.hs b/src/Control/Monad/Trans/Iter.hs
index 25c8239..7a46ce2 100644
--- a/src/Control/Monad/Trans/Iter.hs
+++ b/src/Control/Monad/Trans/Iter.hs
@@ -219,8 +219,10 @@ instance Monad m => Monad (IterT m) where
   {-# INLINE return #-}
   IterT m >>= k = IterT $ m >>= either (runIterT . k) (return . Right . (>>= k))
   {-# INLINE (>>=) #-}
+#if !(MIN_VERSION_base(4,13,0))
   fail = Fail.fail
   {-# INLINE fail #-}
+#endif
 
 instance Monad m => Fail.MonadFail (IterT m) where
   fail _ = never
