diff -ru gloss-rendering-1.11.1.1.orig/gloss-rendering.cabal gloss-rendering-1.11.1.1/gloss-rendering.cabal
--- gloss-rendering-1.11.1.1.orig/gloss-rendering.cabal	2018-01-09 00:03:55.457742537 -0500
+++ gloss-rendering-1.11.1.1/gloss-rendering.cabal	2018-01-09 00:09:54.205751572 -0500
@@ -34,7 +34,8 @@
         bytestring == 0.10.*,
         OpenGL     >= 2.12 && < 3.1,
         GLUT       == 2.7.*,
-        bmp        == 1.2.*
+        bmp        == 1.2.*,
+        semigroups >= 0.16 && < 0.19
 
   ghc-options:
         -Wall -O2
diff -ru gloss-rendering-1.11.1.1.orig/Graphics/Gloss/Internals/Data/Picture.hs gloss-rendering-1.11.1.1/Graphics/Gloss/Internals/Data/Picture.hs
--- gloss-rendering-1.11.1.1.orig/Graphics/Gloss/Internals/Data/Picture.hs	2017-03-13 23:24:47.000000000 -0400
+++ gloss-rendering-1.11.1.1/Graphics/Gloss/Internals/Data/Picture.hs	2018-01-09 00:10:42.741752794 -0500
@@ -23,7 +23,8 @@
 import Foreign.Marshal.Utils
 import Foreign.Ptr
 import Data.Word
-import Data.Monoid
+import Data.Monoid (Monoid(..))
+import Data.Semigroup (Semigroup(..))
 import Data.ByteString
 import Data.Data
 import System.IO.Unsafe
@@ -125,9 +126,12 @@
 
 
 -- Instances ------------------------------------------------------------------
+instance Semigroup Picture where
+        a <> b          = Pictures [a, b]
+
 instance Monoid Picture where
         mempty          = Blank
-        mappend a b     = Pictures [a, b]
+        mappend         = (<>)
         mconcat         = Pictures
 
 
