diff -ru happy-1.19.9.orig/src/AttrGrammarParser.hs happy-1.19.9/src/AttrGrammarParser.hs
--- happy-1.19.9.orig/src/AttrGrammarParser.hs  2018-02-04 11:17:42.000000000 -0500
+++ happy-1.19.9/src/AttrGrammarParser.hs       2019-03-15 09:32:42.173433910 -0400
@@ -339,7 +339,7 @@


 happyError :: P a
-happyError = fail ("Parse error\n")
+happyError = error ("Parse error\n")
 {-# LINE 1 "templates/GenericTemplate.hs" #-}
 {-# LINE 1 "templates/GenericTemplate.hs" #-}
 {-# LINE 1 "<built-in>" #-}
diff -ru happy-1.19.9.orig/src/Lexer.lhs happy-1.19.9/src/Lexer.lhs
--- happy-1.19.9.orig/src/Lexer.lhs     2018-02-04 11:17:42.000000000 -0500
+++ happy-1.19.9/src/Lexer.lhs  2019-03-15 09:32:31.213371494 -0400
@@ -240,7 +240,7 @@
 > lexReadString []           fn = fn "" []

 > lexError :: String -> String -> Int -> ParseResult a
-> lexError err = runP (lineP >>= \l -> fail (show l ++ ": " ++ err ++ "\n"))
+> lexError err = runP (lineP >>= \l -> error (show l ++ ": " ++ err ++ "\n"))

 > lexNestedComment :: Int -> ([Char] -> Int -> ParseResult a) -> [Char] -> Int
 >                  -> ParseResult a
diff -ru happy-1.19.9.orig/src/Parser.hs happy-1.19.9/src/Parser.hs
--- happy-1.19.9.orig/src/Parser.hs     2018-02-04 11:17:42.000000000 -0500
+++ happy-1.19.9/src/Parser.hs  2019-03-15 09:32:57.125519084 -0400
@@ -736,7 +736,7 @@


 happyError :: P a
-happyError = lineP >>= \l -> fail (show l ++ ": Parse error\n")
+happyError = lineP >>= \l -> error (show l ++ ": Parse error\n")
 {-# LINE 1 "templates/GenericTemplate.hs" #-}
 {-# LINE 1 "templates/GenericTemplate.hs" #-}
 {-# LINE 1 "<built-in>" #-}
