diff -ru hashable-1.2.7.0.orig/Data/Hashable/Generic.hs hashable-1.2.7.0/Data/Hashable/Generic.hs
--- hashable-1.2.7.0.orig/Data/Hashable/Generic.hs	2018-03-07 17:02:09.000000000 -0500
+++ hashable-1.2.7.0/Data/Hashable/Generic.hs	2018-06-17 14:31:55.858697963 -0400
@@ -1,6 +1,9 @@
 {-# LANGUAGE BangPatterns, FlexibleInstances, KindSignatures,
              ScopedTypeVariables, TypeOperators,
-             MultiParamTypeClasses, GADTs, FlexibleContexts #-}
+             MultiParamTypeClasses, GADTs, FlexibleContexts, CPP #-}
+#if __GLASGOW_HASKELL__ >= 805
+{-# LANGUAGE StarIsType #-}
+#endif
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 
 ------------------------------------------------------------------------
