diff -ru hslogger-1.2.10.orig/src/System/Log/Logger.hs hslogger-1.2.10/src/System/Log/Logger.hs
--- hslogger-1.2.10.orig/src/System/Log/Logger.hs	2016-05-29 09:42:33.000000000 -0400
+++ hslogger-1.2.10/src/System/Log/Logger.hs	2018-07-04 21:47:31.436345560 -0400
@@ -476,7 +476,7 @@
 removeAllHandlers :: IO ()
 removeAllHandlers =
     modifyMVar_ logTree $ \lt -> do
-        let allHandlers = Map.fold (\l r -> concat [r, handlers l]) [] lt
+        let allHandlers = Map.foldr (\l r -> concat [r, handlers l]) [] lt
         mapM_ (\(HandlerT h) -> close h) allHandlers
         return $ Map.map (\l -> l {handlers = []}) lt
 
