diff -ru hspec-core-2.4.4.orig/src/Test/Hspec/Core/Compat.hs hspec-core-2.4.4/src/Test/Hspec/Core/Compat.hs
--- hspec-core-2.4.4.orig/src/Test/Hspec/Core/Compat.hs	2017-06-16 11:08:23.000000000 +0200
+++ hspec-core-2.4.4/src/Test/Hspec/Core/Compat.hs	2017-09-17 12:30:59.440828450 +0200
@@ -12,6 +12,7 @@
 , module Data.Foldable
 , module Data.Traversable
 , module Data.Monoid
+, module Data.Semigroup
 
 #if !MIN_VERSION_base(4,6,0)
 , modifyIORef'
@@ -21,7 +22,8 @@
 import           Control.Applicative
 import           Data.Foldable
 import           Data.Traversable
-import           Data.Monoid
+import           Data.Monoid hiding ((<>))
+import           Data.Semigroup (Semigroup(..))
 
 import           Prelude hiding (
     all
Only in hspec-core-2.4.4/src/Test/Hspec/Core: Compat.hs~
diff -ru hspec-core-2.4.4.orig/src/Test/Hspec/Core/Runner.hs hspec-core-2.4.4/src/Test/Hspec/Core/Runner.hs
--- hspec-core-2.4.4.orig/src/Test/Hspec/Core/Runner.hs	2017-06-16 11:08:23.000000000 +0200
+++ hspec-core-2.4.4/src/Test/Hspec/Core/Runner.hs	2017-09-17 12:31:26.856678058 +0200
@@ -226,6 +226,9 @@
 , summaryFailures :: Int
 } deriving (Eq, Show)
 
+instance Semigroup Summary where
+  (Summary x1 x2) <> (Summary y1 y2) = Summary (x1 + y1) (x2 + y2)
+
 instance Monoid Summary where
   mempty = Summary 0 0
-  (Summary x1 x2) `mappend` (Summary y1 y2) = Summary (x1 + y1) (x2 + y2)
+  mappend = (<>)
Only in hspec-core-2.4.4/src/Test/Hspec/Core: Runner.hs~
