diff --git a/hspec-core/src/Test/Hspec/Core/Runner.hs b/hspec-core/src/Test/Hspec/Core/Runner.hs
index dfc2c6e..e484681 100644
--- a/hspec-core/src/Test/Hspec/Core/Runner.hs
+++ b/hspec-core/src/Test/Hspec/Core/Runner.hs
@@ -226,6 +226,9 @@ data Summary = Summary {
 , summaryFailures :: Int
 } deriving (Eq, Show)
 
+instance Semigroup Summary where
+  (<>) = mappend
+
 instance Monoid Summary where
   mempty = Summary 0 0
   (Summary x1 x2) `mappend` (Summary y1 y2) = Summary (x1 + y1) (x2 + y2)
