diff -ru http-api-data-0.3.7.1.orig/src/Web/Internal/FormUrlEncoded.hs http-api-data-0.3.7.1/src/Web/Internal/FormUrlEncoded.hs
--- http-api-data-0.3.7.1.orig/src/Web/Internal/FormUrlEncoded.hs	2016-11-14 21:28:23.000000000 +0100
+++ http-api-data-0.3.7.1/src/Web/Internal/FormUrlEncoded.hs	2017-09-15 11:52:48.979381801 +0200
@@ -55,6 +55,10 @@
 import Numeric.Natural
 #endif
 
+#if !(MIN_VERSION_base(4,11,0))
+import Data.Semigroup
+#endif
+
 import           GHC.Exts                   (IsList (..), Constraint)
 import           GHC.Generics
 import           GHC.TypeLits
@@ -182,7 +186,7 @@
 --
 -- 'Form' can be URL-encoded with 'urlEncodeForm' and URL-decoded with 'urlDecodeForm'.
 newtype Form = Form { unForm :: HashMap Text [Text] }
-  deriving (Eq, Read, Generic, Monoid)
+  deriving (Eq, Read, Generic, Semigroup, Monoid)
 
 instance Show Form where
   showsPrec d form = showParen (d > 10) $
Only in http-api-data-0.3.7.1/src/Web/Internal: FormUrlEncoded.hs~
