diff -ru hxt-9.3.1.16.orig/src/Text/XML/HXT/Arrow/Pickle/Xml.hs hxt-9.3.1.16/src/Text/XML/HXT/Arrow/Pickle/Xml.hs
--- hxt-9.3.1.16.orig/src/Text/XML/HXT/Arrow/Pickle/Xml.hs	2016-08-10 03:55:56.000000000 -0400
+++ hxt-9.3.1.16/src/Text/XML/HXT/Arrow/Pickle/Xml.hs	2019-03-15 15:07:12.856241363 -0400
@@ -65,6 +65,7 @@
 import           Control.Monad.Error              (MonadError (..))
 #endif
 
+import qualified Control.Monad.Fail               as Fail
 import           Control.Monad.State              (MonadState (..), gets,
                                                    modify)
 
@@ -141,8 +142,12 @@
                   case r of
                     Left err -> (Left err, st')
                     Right v  -> runUP (f v) st'
-    fail        = throwMsg                              -- don't use fail, use throwError
+#if !(MIN_VERSION_base(4,13,0))
+    fail        = Fail.fail
+#endif
 
+instance Fail.MonadFail Unpickler where
+    fail        = throwMsg                              -- don't use fail, use throwError
 
 instance MonadState St Unpickler where
     get         = UP $ \ st -> (Right st, st)
