commit c9daaeb85a1fcd50af24e4fba329670a9ba379b6
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Sat Oct 27 21:47:44 2018 -0400

    Allow building with template-haskell-2.15.0.0

diff --git a/Control/Exception/IOChoice/THUtil.hs b/Control/Exception/IOChoice/THUtil.hs
index 46e63f0..f342545 100644
--- a/Control/Exception/IOChoice/THUtil.hs
+++ b/Control/Exception/IOChoice/THUtil.hs
@@ -24,8 +24,16 @@ checkSupported exc = do
 #if __GLASGOW_HASKELL__ >= 800
         DataD _ name [] _ _ _ -> conT name
         NewtypeD _ name [] _ _ _ -> conT name
-        DataInstD _ name args _ _ _ -> foldl1 appT (conT name:map return args)
-        NewtypeInstD _ name args _ _ _ -> foldl1 appT (conT name:map return args)
+        DataInstD _ name
+# if MIN_VERSION_template_haskell(2,15,0)
+                         _
+# endif
+                         args _ _ _ -> foldl1 appT (conT name:map return args)
+        NewtypeInstD _ name
+# if MIN_VERSION_template_haskell(2,15,0)
+                            _
+# endif
+                            args _ _ _ -> foldl1 appT (conT name:map return args)
 #else
         DataD _ name [] _ _ -> conT name
         NewtypeD _ name [] _ _ -> conT name
@@ -34,7 +42,11 @@ checkSupported exc = do
 #endif
         TySynD name [] _ -> conT name
 #if __GLASGOW_HASKELL__ >= 707
-        TySynInstD name (TySynEqn _ t) -> foldl1 appT (conT name:[return t])
+        TySynInstD name (TySynEqn
+# if MIN_VERSION_template_haskell(2,15,0)
+                                  _
+# endif
+                                  _ t) -> foldl1 appT (conT name:[return t])
 #else
         TySynInstD name args _ -> foldl1 appT (conT name:map return args)
 #endif
