diff -ru JuicyPixels-3.2.9.1/src/Codec/Picture/Metadata.hs JuicyPixels-3.2.9.1-patched/src/Codec/Picture/Metadata.hs
--- JuicyPixels-3.2.9.1/src/Codec/Picture/Metadata.hs	2017-11-11 17:35:34.000000000 +0800
+++ JuicyPixels-3.2.9.1-patched/src/Codec/Picture/Metadata.hs	2017-12-08 12:22:34.000000000 +0800
@@ -46,7 +46,7 @@
 import Data.Monoid( Monoid, mempty, mappend )
 import Data.Word( Word )
 #endif
-
+import Data.Semigroup( Semigroup(..) )
 
 import Control.DeepSeq( NFData( .. ) )
 import qualified Data.Foldable as F
@@ -169,9 +169,12 @@
   }
   deriving (Show, NFData)
 
+instance Semigroup Metadatas where
+  (<>) = union
+
 instance Monoid Metadatas where
   mempty = empty
-  mappend = union
+  mappend = (<>)
 
 -- | Right based union
 union :: Metadatas -> Metadatas -> Metadatas
