diff -ru language-c-0.8.1.orig/src/Language/C/Analysis/DefTable.hs language-c-0.8.1/src/Language/C/Analysis/DefTable.hs
--- language-c-0.8.1.orig/src/Language/C/Analysis/DefTable.hs	2018-06-08 00:48:29.000000000 -0400
+++ language-c-0.8.1/src/Language/C/Analysis/DefTable.hs	2018-07-04 21:30:47.308320272 -0400
@@ -106,7 +106,7 @@
 
 -- | get the globally defined entries of a definition table
 globalDefs :: DefTable -> GlobalDecls
-globalDefs deftbl = Map.foldWithKey insertDecl (GlobalDecls e gtags e) (globalNames $ identDecls deftbl)
+globalDefs deftbl = Map.foldrWithKey insertDecl (GlobalDecls e gtags e) (globalNames $ identDecls deftbl)
     where
     e = Map.empty
     (_fwd_decls,gtags) = Map.mapEither id $ globalNames (tagDecls deftbl)
diff -ru language-c-0.8.1.orig/src/Language/C/Analysis/SemRep.hs language-c-0.8.1/src/Language/C/Analysis/SemRep.hs
--- language-c-0.8.1.orig/src/Language/C/Analysis/SemRep.hs	2018-06-08 00:48:29.000000000 -0400
+++ language-c-0.8.1/src/Language/C/Analysis/SemRep.hs	2018-07-04 21:30:29.036319812 -0400
@@ -136,7 +136,7 @@
                                                 ( Map Ident Enumerator,
                                                   Map Ident ObjDef,
                                                   Map Ident FunDef ) )
-splitIdentDecls include_all = Map.foldWithKey (if include_all then deal else deal') (Map.empty,(Map.empty,Map.empty,Map.empty))
+splitIdentDecls include_all = Map.foldrWithKey (if include_all then deal else deal') (Map.empty,(Map.empty,Map.empty,Map.empty))
   where
   deal ident entry (decls,defs) = (Map.insert ident (declOfDef entry) decls, addDef ident entry defs)
   deal' ident (Declaration d) (decls,defs) = (Map.insert ident d decls,defs)
