diff -ru language-c-0.8.2.orig/src/Language/C/Parser/ParserMonad.hs language-c-0.8.2/src/Language/C/Parser/ParserMonad.hs
--- language-c-0.8.2.orig/src/Language/C/Parser/ParserMonad.hs	2018-09-19 00:21:29.000000000 -0400
+++ language-c-0.8.2/src/Language/C/Parser/ParserMonad.hs	2019-04-19 16:43:53.153364757 -0400
@@ -48,6 +48,7 @@
 
 import Control.Applicative (Applicative(..))
 import Control.Monad (liftM, ap)
+import qualified Control.Monad.Fail as Fail
 import Data.Set  (Set)
 import qualified Data.Set as Set (fromList, insert, member, delete)
 
@@ -82,8 +83,13 @@
 instance Monad P where
   return = returnP
   (>>=) = thenP
-  fail m = getPos >>= \pos -> failP pos [m]
+#if !(MIN_VERSION_base(4,13,0))
+  fail = Fail.fail
+#endif
 
+instance Fail.MonadFail P where
+  fail m = getPos >>= \pos -> failP pos [m]
+  
 
 -- | execute the given parser on the supplied input stream.
 --   returns 'ParseError' if the parser failed, and a pair of
