diff -ru List-0.6.0.orig/src/Control/Monad/ListT.hs List-0.6.0/src/Control/Monad/ListT.hs
--- List-0.6.0.orig/src/Control/Monad/ListT.hs	2016-10-31 09:42:01.000000000 +0100
+++ List-0.6.0/src/Control/Monad/ListT.hs	2017-09-17 11:20:22.403495565 +0200
@@ -30,6 +30,7 @@
 import Control.Monad.IO.Class (MonadIO(..))
 import Control.Monad.Trans.Class (MonadTrans(..))
 import Data.Monoid (Monoid(..))
+import Data.Semigroup
 
 newtype ListT m a =
     ListT { runListT :: m (ListItem (ListT m) a) }
@@ -46,9 +47,12 @@
     where
         step x = return . consFunc x . joinL
 
+instance Monad m => Semigroup (ListT m a) where
+    (<>) = flip (foldrL' cons)
+
 instance Monad m => Monoid (ListT m a) where
     mempty = ListT $ return Nil
-    mappend = flip (foldrL' cons)
+    mappend = (<>)
 
 instance Monad m => Functor (ListT m) where
     fmap func = foldrL' (cons . func) mempty
Only in List-0.6.0/src/Control/Monad: ListT.hs~
