diff -ru managed-1.0.5.orig/src/Control/Monad/Managed.hs managed-1.0.5/src/Control/Monad/Managed.hs
--- managed-1.0.5.orig/src/Control/Monad/Managed.hs	2016-05-29 02:12:29.000000000 +0200
+++ managed-1.0.5/src/Control/Monad/Managed.hs	2017-09-14 22:31:21.404647462 +0200
@@ -110,6 +110,10 @@
 import Control.Monad.IO.Class (MonadIO(liftIO))
 import Control.Monad.Trans.Class (lift)
 
+#if MIN_VERSION_base(4,9,0)
+import Data.Semigroup
+#endif
+
 #if MIN_VERSION_base(4,8,0)
 import Control.Applicative (liftA2)
 #else
@@ -162,10 +166,12 @@
         a <- m
         return_ a )
 
+instance Monoid a => Semigroup (Managed a) where
+    (<>) = liftA2 mappend
+
 instance Monoid a => Monoid (Managed a) where
     mempty = pure mempty
-
-    mappend = liftA2 mappend
+    mappend = (<>)
 
 instance Num a => Num (Managed a) where
     fromInteger = pure . fromInteger
Only in managed-1.0.5/src/Control/Monad: Managed.hs~
