diff -ru microlens-0.4.8.1.orig/src/Lens/Micro.hs microlens-0.4.8.1/src/Lens/Micro.hs
--- microlens-0.4.8.1.orig/src/Lens/Micro.hs	2017-08-11 20:23:02.000000000 +0200
+++ microlens-0.4.8.1/src/Lens/Micro.hs	2017-09-17 12:39:28.310024297 +0200
@@ -1270,10 +1270,14 @@
 
 newtype Traversed a f = Traversed { getTraversed :: f a }
 
+instance Applicative f => Semigroup (Traversed a f) where
+  Traversed ma <> Traversed mb = Traversed (ma *> mb)
+  {-# INLINE (<>) #-}
+
 instance Applicative f => Monoid (Traversed a f) where
   mempty = Traversed (pure (error "Lens.Micro.Traversed: value used"))
   {-# INLINE mempty #-}
-  Traversed ma `mappend` Traversed mb = Traversed (ma *> mb)
+  mappend = (<>)
   {-# INLINE mappend #-}
 
 newtype Bazaar a b t = Bazaar (forall f. Applicative f => (a -> f b) -> f t)
Only in microlens-0.4.8.1/src/Lens: Micro.hs~
