commit 43567905fcdebcaf781a194ad0a006a8053dec82
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Thu May 2 10:59:53 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/src/Graphics/Rendering/OpenGL/GL/IOState.hs b/src/Graphics/Rendering/OpenGL/GL/IOState.hs
index c81c4c0..3ec5cee 100644
--- a/src/Graphics/Rendering/OpenGL/GL/IOState.hs
+++ b/src/Graphics/Rendering/OpenGL/GL/IOState.hs
@@ -40,7 +40,9 @@ instance Functor (IOState s) where
 instance Monad (IOState s) where
    return a = IOState $ \s -> return (a, s)
    m >>= k  = IOState $ \s -> do (a, s') <- runIOState m s ; runIOState (k a) s'
+#if !(MIN_VERSION_base(4,13,0))
    fail str = IOState $ \_ -> fail str
+#endif
 
 getIOState :: IOState s (Ptr s)
 getIOState = IOState $ \s -> return (s, s)
