diff -ru optional-args-1.0.1.orig/src/Data/Optional.hs optional-args-1.0.1/src/Data/Optional.hs
--- optional-args-1.0.1.orig/src/Data/Optional.hs	2016-04-16 06:11:21.000000000 +0200
+++ optional-args-1.0.1/src/Data/Optional.hs	2017-09-14 22:38:35.114394135 +0200
@@ -97,6 +97,7 @@
 import Data.Foldable (Foldable)
 import Data.Traversable (Traversable)
 import Data.Monoid (Monoid(..))
+import Data.Semigroup (Semigroup(..))
 import Data.String (IsString(..))
 
 -- | A function argument that has a `Default` value
@@ -125,10 +126,13 @@
     mzero = empty
     mplus = (<|>)
 
+-- TODO: can be more general but needs CPP
+instance Monoid a => Semigroup (Optional a) where
+    (<>) = liftA2 mappend
+
 instance Monoid a => Monoid (Optional a) where
     mempty = pure mempty
-
-    mappend = liftA2 mappend
+    mappend = (<>)
 
 instance IsString a => IsString (Optional a) where
     fromString str = pure (fromString str)
Only in optional-args-1.0.1/src/Data: Optional.hs~
