diff -ru persistent-template-2.7.0.orig/Database/Persist/TH.hs persistent-template-2.7.0/Database/Persist/TH.hs
--- persistent-template-2.7.0.orig/Database/Persist/TH.hs	2019-04-15 22:46:49.000000000 -0400
+++ persistent-template-2.7.0/Database/Persist/TH.hs	2019-04-19 17:46:14.353877721 -0400
@@ -502,8 +502,14 @@
 
 uniqueTypeDec :: MkPersistSettings -> EntityDef -> Dec
 uniqueTypeDec mps t =
-    DataInstD [] ''Unique
+    DataInstD []
+#if MIN_VERSION_template_haskell(2,15,0)
+        Nothing
+        (ConT ''Unique `AppT` genericDataType mps (entityHaskell t) backendT)
+#else
+        ''Unique
         [genericDataType mps (entityHaskell t) backendT]
+#endif
             Nothing
             (map (mkUnique mps t) $ entityUniques t)
             (derivClause $ entityUniques t)
@@ -765,7 +771,12 @@
                         bi <- backendKeyI
                         return (bi, allInstances)
 
-#if MIN_VERSION_template_haskell(2,12,0)
+#if MIN_VERSION_template_haskell(2,15,0)
+    cxti <- mapM conT i
+    let kd = if useNewtype
+               then NewtypeInstD [] Nothing (ConT k `AppT` recordType) Nothing dec [DerivClause Nothing cxti]
+               else DataInstD    [] Nothing (ConT k `AppT` recordType) Nothing [dec] [DerivClause Nothing cxti]
+#elif MIN_VERSION_template_haskell(2,12,0)
     cxti <- mapM conT i
     let kd = if useNewtype
                then NewtypeInstD [] k [recordType] Nothing dec [DerivClause Nothing cxti]
@@ -1052,19 +1063,34 @@
         , puk
         , DataInstD
             []
+#if MIN_VERSION_template_haskell(2,15,0)
+            Nothing
+            (ConT ''EntityField `AppT`
+             genDataType `AppT`
+             (VarT $ mkName "typ"))
+#else
             ''EntityField
             [ genDataType
             , VarT $ mkName "typ"
             ]
+#endif
             Nothing
             (map fst fields)
             []
         , FunD 'persistFieldDef (map snd fields)
+#if MIN_VERSION_template_haskell(2,15,0)
+        , TySynInstD
+            (TySynEqn
+               Nothing
+               (ConT ''PersistEntityBackend `AppT` genDataType)
+               (backendDataType mps))
+#else
         , TySynInstD
             ''PersistEntityBackend
             (TySynEqn
                [genDataType]
                (backendDataType mps))
+#endif
         , FunD 'persistIdField [normalClause [] (ConE $ keyIdName t)]
         , FunD 'fieldLens lensClauses
         ]
