diff -ru persistent-2.10.0.orig/Database/Persist/Types/Base.hs persistent-2.10.0/Database/Persist/Types/Base.hs
--- persistent-2.10.0.orig/Database/Persist/Types/Base.hs	2019-04-14 22:27:15.000000000 -0400
+++ persistent-2.10.0/Database/Persist/Types/Base.hs	2019-04-19 16:55:07.211353215 -0400
@@ -4,6 +4,7 @@
 
 import Control.Arrow (second)
 import Control.Exception (Exception)
+import qualified Control.Monad.Fail as Fail
 import Control.Monad.Trans.Error (Error (..))
 import qualified Data.Aeson as A
 import Data.Bits (shiftL, shiftR)
@@ -433,10 +434,10 @@
     parseJSON (A.String t0) =
         case T.uncons t0 of
             Nothing -> fail "Null string"
-            Just ('p', t) -> either (fail "Invalid base64") (return . PersistDbSpecific)
+            Just ('p', t) -> either (\_ -> fail "Invalid base64") (return . PersistDbSpecific)
                            $ B64.decode $ TE.encodeUtf8 t
             Just ('s', t) -> return $ PersistText t
-            Just ('b', t) -> either (fail "Invalid base64") (return . PersistByteString)
+            Just ('b', t) -> either (\_ -> fail "Invalid base64") (return . PersistByteString)
                            $ B64.decode $ TE.encodeUtf8 t
             Just ('t', t) -> fmap PersistTimeOfDay $ readMay t
             Just ('u', t) -> fmap PersistUTCTime $ readMay t
@@ -448,7 +449,7 @@
       where
         headMay []    = Nothing
         headMay (x:_) = Just x
-        readMay :: (Read a, Monad m) => T.Text -> m a
+        readMay :: (Read a, Fail.MonadFail m) => T.Text -> m a
         readMay t =
             case reads $ T.unpack t of
                 (x, _):_ -> return x
