diff -ru pipes-4.3.9.orig/src/Pipes.hs pipes-4.3.9/src/Pipes.hs
--- pipes-4.3.9.orig/src/Pipes.hs	2018-03-12 14:27:03.000000000 -0400
+++ pipes-4.3.9/src/Pipes.hs	2019-03-16 08:40:31.569137565 -0400
@@ -66,6 +66,7 @@
 import Control.Monad (void, MonadPlus(mzero, mplus))
 import Control.Monad.Catch (MonadThrow(..), MonadCatch(..))
 import Control.Monad.Except (MonadError(..))
+import qualified Control.Monad.Fail as Fail
 import Control.Monad.IO.Class (MonadIO(liftIO))
 import Control.Monad.Reader (MonadReader(..))
 import Control.Monad.State (MonadState(..))
@@ -423,6 +424,12 @@
     {-# INLINE return #-}
     m >>= f  = Select (for (enumerate m) (\a -> enumerate (f a)))
     {-# INLINE (>>=) #-}
+#if !(MIN_VERSION_base(4,13,0))
+    fail     = Fail.fail
+    {-# INLINE fail #-}
+#endif
+
+instance Monad m => Fail.MonadFail (ListT m) where
     fail _   = mzero
     {-# INLINE fail #-}
 
