diff -ru protolude-0.2.2.orig/src/Protolude/Base.hs protolude-0.2.2/src/Protolude/Base.hs
--- protolude-0.2.2.orig/src/Protolude/Base.hs	2018-02-05 11:35:26.000000000 -0500
+++ protolude-0.2.2/src/Protolude/Base.hs	2018-07-04 21:24:34.620310887 -0400
@@ -122,8 +122,10 @@
 
 #if ( __GLASGOW_HASKELL__ >= 800 )
 import Data.Kind as X (
-    type (*)
-  , type Type
+# if __GLASGOW_HASKELL__ < 805
+    type (*),
+# endif
+    type Type
   )
 #endif
 
diff -ru protolude-0.2.2.orig/src/Protolude.hs protolude-0.2.2/src/Protolude.hs
--- protolude-0.2.2.orig/src/Protolude.hs	2018-03-26 06:36:35.000000000 -0400
+++ protolude-0.2.2/src/Protolude.hs	2018-07-04 21:24:25.504310657 -0400
@@ -492,8 +492,10 @@
 import Control.Monad.STM as X (
     STM
   , atomically
+#if !(MIN_VERSION_stm(2,5,0))
   , always
   , alwaysSucceeds
+#endif
   , retry
   , orElse
   , check
