commit 163fe274fb0eef35fd2a22f6601e09dfcda977c9
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Sat Mar 16 11:20:05 2019 -0400

    Adapt to template-haskell-2.15.0.0

diff --git a/src/Data/Functor/Foldable/TH.hs b/src/Data/Functor/Foldable/TH.hs
index 076f0ed..8041a86 100644
--- a/src/Data/Functor/Foldable/TH.hs
+++ b/src/Data/Functor/Foldable/TH.hs
@@ -194,7 +194,9 @@ makePrimForDI' rules isNewtype tyName vars cons = do
 #endif
 
     -- type instance Base
-#if MIN_VERSION_template_haskell(2,9,0)
+#if MIN_VERSION_template_haskell(2,15,0)
+    let baseDec = TySynInstD (TySynEqn Nothing (ConT baseTypeName `AppT` s) $ conAppsT tyNameF vars')
+#elif MIN_VERSION_template_haskell(2,9,0)
     let baseDec = TySynInstD baseTypeName (TySynEqn [s] $ conAppsT tyNameF vars')
 #else
     let baseDec = TySynInstD baseTypeName [s] $ conAppsT tyNameF vars'
