diff -ru regex-tdfa-1.2.3.2.orig/Text/Regex/TDFA.hs regex-tdfa-1.2.3.2/Text/Regex/TDFA.hs
--- regex-tdfa-1.2.3.2.orig/Text/Regex/TDFA.hs	2019-05-08 17:56:36.000000000 -0400
+++ regex-tdfa-1.2.3.2/Text/Regex/TDFA.hs	2019-05-11 09:11:57.955453593 -0400
@@ -176,6 +176,7 @@
                       ,module Text.Regex.TDFA.Common
                       ,module Text.Regex.Base) where
 
+import qualified Control.Monad.Fail as Fail
 import Data.Version(Version)
 import Text.Regex.Base
 import Text.Regex.TDFA.String()
@@ -202,9 +203,9 @@
 
 -- | This is the monadic matching operator.  If a single match fails,
 -- then 'fail' will be called.
-(=~~) :: (RegexMaker Regex CompOption ExecOption source,RegexContext Regex source1 target,Monad m)
+(=~~) :: (RegexMaker Regex CompOption ExecOption source,RegexContext Regex source1 target,Fail.MonadFail m)
       => source1 -> source -> m target
-(=~~) x r = do let make :: (RegexMaker Regex CompOption ExecOption a, Monad m) => a -> m Regex
+(=~~) x r = do let make :: (RegexMaker Regex CompOption ExecOption a, Fail.MonadFail m) => a -> m Regex
                    make = makeRegexM
                q <- make r
                matchM q x
