diff -ru repa-examples-3.4.1.1.orig/repa-examples.cabal repa-examples-3.4.1.1/repa-examples.cabal
--- repa-examples-3.4.1.1.orig/repa-examples.cabal	2016-06-18 01:50:49.000000000 -0400
+++ repa-examples-3.4.1.1/repa-examples.cabal	2018-06-22 13:19:58.997468489 -0400
@@ -35,11 +35,13 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-mmult
@@ -57,13 +59,15 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-laplace
@@ -80,11 +84,13 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-fft2d
@@ -100,13 +106,15 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-fft2d-highpass
@@ -122,13 +130,15 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-fft3d-highpass
@@ -143,13 +153,15 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-blur
@@ -165,12 +177,15 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph -fno-liberate-case
+        -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-sobel
@@ -186,13 +201,15 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-volume
@@ -206,13 +223,15 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
 
 Executable repa-unit-test
@@ -227,11 +246,13 @@
         -rtsopts
         -threaded
         -eventlog
-        -Odph
         -fno-liberate-case
         -funfolding-use-threshold100
         -funfolding-keeness-factor100
  if flag(llvm)
   ghc-options:
         -fllvm -optlo-O3
+ if !impl(ghc >= 8.5)
+  ghc-options:
+        -Odph
 
