commit b0653b482ed95b67ac793c97d95c5af28ced8e69
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Fri Mar 15 18:34:24 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/Web/Scotty/Internal/Types.hs b/Web/Scotty/Internal/Types.hs
index e0c0d56..eb58bbd 100644
--- a/Web/Scotty/Internal/Types.hs
+++ b/Web/Scotty/Internal/Types.hs
@@ -141,7 +141,9 @@ newtype ActionT e m a = ActionT { runAM :: ExceptT (ActionError e) (ReaderT Acti
 instance (Monad m, ScottyError e) => Monad (ActionT e m) where
     return = ActionT . return
     ActionT m >>= k = ActionT (m >>= runAM . k)
+#if !(MIN_VERSION_base(4,13,0))
     fail = ActionT . throwError . stringError
+#endif
 
 instance (Monad m, ScottyError e) => Fail.MonadFail (ActionT e m) where
     fail = ActionT . throwError . stringError
