diff --git a/semigroupoids.cabal b/semigroupoids.cabal
index 035eba6..7cb5667 100644
--- a/semigroupoids.cabal
+++ b/semigroupoids.cabal
@@ -152,7 +152,7 @@ library
     build-depends: ghc-prim
 
   if flag(containers)
-    build-depends: containers >= 0.3 && < 0.6
+    build-depends: containers >= 0.3 && < 0.7
 
   if flag(contravariant)
     build-depends: contravariant >= 0.2.0.1 && < 2
