diff -ru shake-0.16.4.orig/src/General/FileLock.hs shake-0.16.4/src/General/FileLock.hs
--- shake-0.16.4.orig/src/General/FileLock.hs	2018-04-04 15:25:53.000000000 -0400
+++ shake-0.16.4/src/General/FileLock.hs	2018-05-20 14:33:58.011899178 -0400
@@ -60,7 +60,7 @@
     createDirectoryRecursive $ takeDirectory file
     tryIO $ writeFile file ""
 
-    bracket (openFd file ReadWrite Nothing defaultFileFlags) closeFd $ \fd -> do
+    bracket (openSimpleFd file ReadWrite) closeFd $ \fd -> do
         let lock = (WriteLock, AbsoluteSeek, 0, 0)
         res <- tryIO $ setLock fd lock
         case res of
@@ -73,4 +73,10 @@
                                Just (pid, _) -> "Shake process ID " ++ show pid ++ " is using this lock.\n") ++
                           show e
 
+#if MIN_VERSION_unix(2,8,0)
+openSimpleFd file mode = openFd file mode defaultFileFlags
+#else
+openSimpleFd file mode = openFd file mode Nothing defaultFileFlags
+#endif
+
 #endif
