commit 5ca1bd8c3ebe15693fca25acf8addac38b0742c5
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Sat Oct 27 22:27:17 2018 -0400

    Allow building with template-haskell-2.15.0.0

diff --git a/src/Data/Store/TH/Internal.hs b/src/Data/Store/TH/Internal.hs
index e0b777d..5f2a468 100644
--- a/src/Data/Store/TH/Internal.hs
+++ b/src/Data/Store/TH/Internal.hs
@@ -355,8 +355,16 @@ getUnboxInfo = do
     return (map (everywhere (id `extT` dequalVarT) . go) insts)
   where
 #if MIN_VERSION_template_haskell(2,11,0)
-    go (NewtypeInstD preds _ [ty] _ con _) = (preds, ty, conToDataCons con)
-    go (DataInstD preds _ [ty] _ cons _) = (preds, ty, concatMap conToDataCons cons)
+    go (NewtypeInstD preds _
+# if MIN_VERSION_template_haskell(2,15,0)
+                           _
+# endif
+                           [ty] _ con _) = (preds, ty, conToDataCons con)
+    go (DataInstD preds _
+# if MIN_VERSION_template_haskell(2,15,0)
+                        _
+# endif
+                        [ty] _ cons _) = (preds, ty, concatMap conToDataCons cons)
 #else
     go (NewtypeInstD preds _ [ty] con _) = (preds, ty, conToDataCons con)
     go (DataInstD preds _ [ty] cons _) = (preds, ty, concatMap conToDataCons cons)
