diff -ru system-filepath-0.4.13.4.orig/lib/Filesystem/Path.hs system-filepath-0.4.13.4/lib/Filesystem/Path.hs
--- system-filepath-0.4.13.4.orig/lib/Filesystem/Path.hs	2015-05-12 09:50:07.000000000 +0200
+++ system-filepath-0.4.13.4/lib/Filesystem/Path.hs	2017-09-14 22:23:05.751270598 +0200
@@ -61,13 +61,17 @@
 import           Data.List (foldl')
 import           Data.Maybe (isJust, isNothing)
 import qualified Data.Monoid as M
+import qualified Data.Semigroup as S
 import qualified Data.Text as T
 
 import           Filesystem.Path.Internal
 
+instance S.Semigroup FilePath where
+	(<>) = append
+
 instance M.Monoid FilePath where
 	mempty = empty
-	mappend = append
+	mappend = (S.<>)
 	mconcat = concat
 
 -------------------------------------------------------------------------------
Only in system-filepath-0.4.13.4/lib/Filesystem: Path.hs~
