diff -ru tasty-1.1.0.2.orig/Test/Tasty/CmdLine.hs tasty-1.1.0.2/Test/Tasty/CmdLine.hs
--- tasty-1.1.0.2.orig/Test/Tasty/CmdLine.hs	2018-02-12 13:46:24.000000000 -0500
+++ tasty-1.1.0.2/Test/Tasty/CmdLine.hs	2018-06-30 18:36:35.559353670 -0400
@@ -8,7 +8,7 @@
   ) where
 
 import Options.Applicative
-import Data.Monoid
+import Data.Monoid (Monoid(..), (<>))
 import Data.Proxy
 import Data.Foldable (foldMap)
 import Prelude  -- Silence AMP and FTP import warnings
diff -ru tasty-1.1.0.2.orig/Test/Tasty/Ingredients/ConsoleReporter.hs tasty-1.1.0.2/Test/Tasty/Ingredients/ConsoleReporter.hs
--- tasty-1.1.0.2.orig/Test/Tasty/Ingredients/ConsoleReporter.hs	2018-06-23 09:22:35.000000000 -0400
+++ tasty-1.1.0.2/Test/Tasty/Ingredients/ConsoleReporter.hs	2018-06-30 18:37:05.631354427 -0400
@@ -41,7 +41,7 @@
 import Data.Char.WCWidth (wcwidth)
 #endif
 import Data.Maybe
-import Data.Monoid
+import Data.Monoid (Monoid(..), Any(..))
 import Data.Typeable
 import Options.Applicative hiding (str)
 import System.IO
diff -ru tasty-1.1.0.2.orig/Test/Tasty/Runners/Reducers.hs tasty-1.1.0.2/Test/Tasty/Runners/Reducers.hs
--- tasty-1.1.0.2.orig/Test/Tasty/Runners/Reducers.hs	2018-05-06 09:28:51.000000000 -0400
+++ tasty-1.1.0.2/Test/Tasty/Runners/Reducers.hs	2018-06-30 18:35:50.635352539 -0400
@@ -41,7 +41,7 @@
 
 module Test.Tasty.Runners.Reducers where
 
-import Data.Monoid
+import Data.Monoid (Monoid(..))
 import Control.Applicative
 import Prelude  -- Silence AMP import warnings
 #if MIN_VERSION_base(4,9,0)
