diff -ru th-expand-syns-0.4.3.0.orig/Language/Haskell/TH/ExpandSyns.hs th-expand-syns-0.4.3.0/Language/Haskell/TH/ExpandSyns.hs
--- th-expand-syns-0.4.3.0.orig/Language/Haskell/TH/ExpandSyns.hs	2017-04-21 02:05:11.000000000 +0200
+++ th-expand-syns-0.4.3.0/Language/Haskell/TH/ExpandSyns.hs	2017-09-17 12:32:12.476427680 +0200
@@ -17,6 +17,7 @@
 import Data.Generics
 import Control.Monad
 import Data.Monoid
+import Data.Semigroup
 import Prelude
 
 -- For ghci
@@ -91,9 +92,13 @@
       sesWarnTypeFamilies = True
     }
 
-  mappend (SynonymExpansionSettings w1) (SynonymExpansionSettings w2) =
+  mappend = (<>)
+
+instance Semigroup SynonymExpansionSettings where
+  (SynonymExpansionSettings w1) <> (SynonymExpansionSettings w2) =
     SynonymExpansionSettings (w1 && w2)
 
+
 -- | Suppresses the warning that type families are unsupported.
 noWarnTypeFamilies :: SynonymExpansionSettings
 noWarnTypeFamilies = mempty { sesWarnTypeFamilies = False }
Only in th-expand-syns-0.4.3.0/Language/Haskell/TH: ExpandSyns.hs~
