commit a6dc0285a37ca46c4d4771e7955491c37777b430
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Sat Mar 9 15:20:22 2019 -0500

    Work around https://github.com/mgsloan/th-orphans/issues/27

diff --git a/src/Language/Haskell/TH/Instances.hs b/src/Language/Haskell/TH/Instances.hs
index bd4d85b..4b2b522 100644
--- a/src/Language/Haskell/TH/Instances.hs
+++ b/src/Language/Haskell/TH/Instances.hs
@@ -517,5 +517,11 @@ deriving instance Typeable Ppr
 deriving instance Typeable Quasi
 #endif
 
+#if __GLASGOW_HASKELL__ >= 809
+-- Work around https://github.com/mgsloan/th-orphans/issues/27
+instance Lift Bytes where
+  lift (Bytes _fp off sz) = [| Bytes (error "Cannot lift ForeignPtr") off sz |]
+#endif
+
 $(reifyManyWithoutInstances ''Lift [''Info, ''Loc] (const True) >>=
   deriveLiftMany)
