diff --git a/src/Text/Trifecta/Parser.hs b/src/Text/Trifecta/Parser.hs
index 26a5c1c..91695da 100644
--- a/src/Text/Trifecta/Parser.hs
+++ b/src/Text/Trifecta/Parser.hs
@@ -177,8 +177,10 @@ instance Monad Parser where
   {-# INLINE (>>=) #-}
   (>>) = (*>)
   {-# INLINE (>>) #-}
+#if !(MIN_VERSION_base(4,13,0))
   fail = Fail.fail
   {-# INLINE fail #-}
+#endif
 
 instance Fail.MonadFail Parser where
   fail s = Parser $ \ _ ee _ _ _ _ -> ee (failed s)
