commit 684d9831d9799d35b2e2af46f6d171d2fd20fa14
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Sat Dec 29 20:49:37 2018 -0500

    Allow building with template-haskell-2.15.0.0

diff --git a/Unsafe/TrueName.hs b/Unsafe/TrueName.hs
index d126ced..e5476ed 100644
--- a/Unsafe/TrueName.hs
+++ b/Unsafe/TrueName.hs
@@ -65,6 +65,11 @@ decNames dec = case dec of
 #endif
 
 #if MIN_VERSION_template_haskell(2,12,0)
+    DataInstD cxt _mbtvs _ntyps _kind cons derivs   ->
+        datatypeNames cxt cons  ++ derivNames derivs
+    NewtypeInstD cxt _mbtvs _ntyps _kind con derivs ->
+        datatypeNames cxt [con] ++ derivNames derivs
+#elif MIN_VERSION_template_haskell(2,12,0)
     DataInstD cxt _name _typs _kind cons derivs   ->
         datatypeNames cxt cons  ++ derivNames derivs
     NewtypeInstD cxt _name _typs _kind con derivs ->
@@ -93,7 +98,11 @@ decNames dec = case dec of
 #endif
 
 #if MIN_VERSION_template_haskell(2,9,0)
-    TySynInstD _ tse -> tseNames tse
+    TySynInstD
+# if !(MIN_VERSION_template_haskell(2,15,0))
+      _
+# endif
+      tse -> tseNames tse
     RoleAnnotD _ _ -> []
 #else
     TySynInstD _ ts t -> (typNames =<< ts) ++ typNames t
@@ -121,7 +130,19 @@ derivNames derivs = predNames =<<
 
 #if MIN_VERSION_template_haskell(2,9,0)
 tseNames :: TySynEqn -> [Name]
+# if MIN_VERSION_template_haskell(2,15,0)
+tseNames (TySynEqn _ nts t) = (typNames =<< ts) ++ typNames t
+  where
+    (_n, ts) = unAppsT nts
+
+    unAppsT :: Type -> (Type, [Type])
+    unAppsT = go []
+      where
+        go xs (AppT l x) = go (x : xs) l
+        go xs ty = (ty, xs)
+# else
 tseNames (TySynEqn ts t) = (typNames =<< ts) ++ typNames t
+# endif
 #endif
 
 predNames :: Pred -> [Name]{- {{{ -}
