diff -ru turtle-1.4.3.orig/src/Turtle/Line.hs turtle-1.4.3/src/Turtle/Line.hs
--- turtle-1.4.3.orig/src/Turtle/Line.hs	2017-09-03 05:32:33.000000000 +0200
+++ turtle-1.4.3/src/Turtle/Line.hs	2017-09-14 22:42:12.429258366 +0200
@@ -60,7 +60,7 @@
 
 -- | A line of text (does not contain newlines).
 newtype Line = Line Text
-  deriving (Eq, Ord, Show, Monoid)
+  deriving (Eq, Ord, Show, Semigroup, Monoid)
 
 instance IsString Line where
   fromString = fromMaybe (throw NewlineForbidden) . textToLine . fromString
Only in turtle-1.4.3/src/Turtle: Line.hs~
diff -ru turtle-1.4.3.orig/src/Turtle/Pattern.hs turtle-1.4.3/src/Turtle/Pattern.hs
--- turtle-1.4.3.orig/src/Turtle/Pattern.hs	2017-09-03 05:32:33.000000000 +0200
+++ turtle-1.4.3/src/Turtle/Pattern.hs	2017-09-14 22:43:00.097008668 +0200
@@ -114,6 +114,7 @@
 import Data.Char
 import Data.List (foldl')
 import Data.Monoid
+import Data.Semigroup (Semigroup(..))
 import Data.String (IsString(..))
 import Data.Text (Text)
 import qualified Data.Text as Text
@@ -123,6 +124,9 @@
 newtype Pattern a = Pattern { runPattern :: StateT Text [] a }
     deriving (Functor, Applicative, Monad, Alternative, MonadPlus)
 
+instance Monoid a => Semigroup (Pattern a) where
+    (<>) = liftA2 mappend
+
 instance Monoid a => Monoid (Pattern a) where
     mempty  = pure mempty
     mappend = liftA2 mappend
Only in turtle-1.4.3/src/Turtle: Pattern.hs~
diff -ru turtle-1.4.3.orig/src/Turtle/Shell.hs turtle-1.4.3/src/Turtle/Shell.hs
--- turtle-1.4.3.orig/src/Turtle/Shell.hs	2017-09-03 05:32:33.000000000 +0200
+++ turtle-1.4.3/src/Turtle/Shell.hs	2017-09-14 22:43:54.392724014 +0200
@@ -82,6 +82,7 @@
 import Data.Foldable (Foldable)
 import qualified Data.Foldable
 import Data.Monoid
+import Data.Semigroup (Semigroup(..))
 import Data.String (IsString(..))
 import Prelude -- Fix redundant import warnings
 
@@ -159,6 +160,9 @@
     fail = Prelude.fail
 #endif
 
+instance Monoid a => Semigroup (Shell a) where
+    (<>) = liftA2 mappend
+
 instance Monoid a => Monoid (Shell a) where
     mempty  = pure mempty
     mappend = liftA2 mappend
Only in turtle-1.4.3/src/Turtle: Shell.hs~
