Only in vault-0.3.0.7/src/Data/Unique: Really.hs~
diff -ru vault-0.3.0.7.orig/src/Data/Vault/ST/ST.hs vault-0.3.0.7/src/Data/Vault/ST/ST.hs
--- vault-0.3.0.7.orig/src/Data/Vault/ST/ST.hs	2017-04-17 21:50:09.000000000 +0200
+++ vault-0.3.0.7/src/Data/Vault/ST/ST.hs	2017-09-14 22:59:30.203780595 +0200
@@ -13,6 +13,7 @@
     ) where
 
 import Data.Monoid (Monoid(..))
+import Data.Semigroup (Semigroup(..))
 import Prelude hiding (lookup)
 import Control.Applicative hiding (empty)
 import Control.Monad.ST
@@ -38,9 +39,12 @@
     Vault
 ------------------------------------------------------------------------------}
 
+instance Semigroup (Vault s) where
+    (<>) = union
+
 instance Monoid (Vault s) where
     mempty = empty
-    mappend = union
+    mappend = (<>)
 
 -- | The empty vault.
 empty :: Vault s
Only in vault-0.3.0.7/src/Data/Vault/ST: ST.hs~
