diff -ru warp-3.2.23.orig/Network/Wai/Handler/Warp/FdCache.hs warp-3.2.23/Network/Wai/Handler/Warp/FdCache.hs
--- warp-3.2.23.orig/Network/Wai/Handler/Warp/FdCache.hs	2017-11-22 04:16:07.000000000 -0500
+++ warp-3.2.23/Network/Wai/Handler/Warp/FdCache.hs	2018-07-15 07:41:10.495001048 -0400
@@ -69,7 +69,11 @@
 
 openFile :: FilePath -> IO Fd
 openFile path = do
-    fd <- openFd path ReadOnly Nothing defaultFileFlags{nonBlock=False}
+    fd <- openFd path ReadOnly
+#if !(MIN_VERSION_unix(2,8,0))
+                 Nothing
+#endif
+                 defaultFileFlags{nonBlock=False}
     setFileCloseOnExec fd
     return fd
 
