diff -ru x509-store-1.6.5.orig/Data/X509/CertificateStore.hs x509-store-1.6.5/Data/X509/CertificateStore.hs
--- x509-store-1.6.5.orig/Data/X509/CertificateStore.hs	2017-06-26 18:12:09.000000000 +0200
+++ x509-store-1.6.5/Data/X509/CertificateStore.hs	2017-09-15 12:18:24.618736311 +0200
@@ -11,6 +11,7 @@
 import Data.Either (rights)
 import Data.List (foldl', isPrefixOf)
 import Data.Monoid
+import Data.Semigroup
 import Data.PEM (pemParseBS, pemContent)
 import Data.X509
 import qualified Data.Map as M
@@ -26,12 +27,14 @@
 data CertificateStore = CertificateStore (M.Map DistinguishedName SignedCertificate)
                       | CertificateStores [CertificateStore]
 
+instance Semigroup CertificateStore where
+    (<>)    (CertificateStores l)  s2@(CertificateStore _) = CertificateStores (l ++ [s2])
+    (<>) s1@(CertificateStore _)   (CertificateStores l)   = CertificateStores ([s1] ++ l)
+    (<>)    (CertificateStores l1) (CertificateStores l2)  = CertificateStores (l1 ++ l2)
+
 instance Monoid CertificateStore where
     mempty  = CertificateStore M.empty
-    mappend s1@(CertificateStore _)   s2@(CertificateStore _) = CertificateStores [s1,s2]
-    mappend    (CertificateStores l)  s2@(CertificateStore _) = CertificateStores (l ++ [s2])
-    mappend s1@(CertificateStore _)   (CertificateStores l)   = CertificateStores ([s1] ++ l)
-    mappend    (CertificateStores l1) (CertificateStores l2)  = CertificateStores (l1 ++ l2)
+    mappend = (<>)
 
 -- | Create a certificate store out of a list of X509 certificate
 makeCertificateStore :: [SignedCertificate] -> CertificateStore
Only in x509-store-1.6.5/Data/X509: CertificateStore.hs~
