diff -ru xmonad-0.14.orig/src/XMonad/Core.hs xmonad-0.14/src/XMonad/Core.hs
--- xmonad-0.14.orig/src/XMonad/Core.hs	2018-07-30 05:12:59.000000000 -0400
+++ xmonad-0.14/src/XMonad/Core.hs	2018-07-30 08:25:03.875705082 -0400
@@ -434,7 +434,11 @@
                 x
  where
     nullStdin = do
-        fd <- openFd "/dev/null" ReadOnly Nothing defaultFileFlags
+        fd <- openFd "/dev/null" ReadOnly
+#if !(MIN_VERSION_unix(2,8,0))
+                     Nothing
+#endif
+                     defaultFileFlags
         dupTo fd stdInput
         closeFd fd
 
