diff -ru yaml-0.8.23.3.orig/Data/Yaml/Parser.hs yaml-0.8.23.3/Data/Yaml/Parser.hs
--- yaml-0.8.23.3.orig/Data/Yaml/Parser.hs	2017-06-16 05:46:12.000000000 +0200
+++ yaml-0.8.23.3/Data/Yaml/Parser.hs	2017-09-17 10:51:54.961189093 +0200
@@ -15,7 +15,9 @@
 import Data.Conduit
 import Data.Conduit.Lift (runWriterC)
 import qualified Data.Map as Map
-#if !MIN_VERSION_base(4,8,0)
+#if MIN_VERSION_base(4,9,0)
+import Data.Semigroup
+#elif !MIN_VERSION_base(4,8,0)
 import Data.Monoid (Monoid (..))
 #endif
 import Data.Text (Text, pack, unpack)
@@ -36,9 +38,13 @@
 instance Alternative YamlParser where
     empty = fail "empty"
     (<|>) = mplus
+instance Semigroup (YamlParser a) where
+    (<>) = mplus
 instance Monoid (YamlParser a) where
     mempty = fail "mempty"
+#if !(MIN_VERSION_base(4,11,0))
     mappend = mplus
+#endif
 instance Monad YamlParser where
     return = pure
     YamlParser f >>= g = YamlParser $ \am ->
