diff -ru yesod-core-1.6.13.orig/src/Yesod/Routes/TH/RenderRoute.hs yesod-core-1.6.13/src/Yesod/Routes/TH/RenderRoute.hs
--- yesod-core-1.6.13.orig/src/Yesod/Routes/TH/RenderRoute.hs	2019-03-17 05:12:08.000000000 -0400
+++ yesod-core-1.6.13/src/Yesod/Routes/TH/RenderRoute.hs	2019-03-17 09:04:44.369924530 -0400
@@ -142,7 +142,11 @@
     cls <- mkRenderRouteClauses ress
     (cons, decs) <- mkRouteCons ress
 #if MIN_VERSION_template_haskell(2,12,0)
+# if MIN_VERSION_template_haskell(2,15,0)
+    did <- DataInstD [] Nothing (ConT ''Route `AppT` typ) Nothing cons <$> fmap (pure . DerivClause Nothing) (mapM conT (clazzes False))
+# else
     did <- DataInstD [] ''Route [typ] Nothing cons <$> fmap (pure . DerivClause Nothing) (mapM conT (clazzes False))
+# endif
     let sds = fmap (\t -> StandaloneDerivD Nothing cxt $ ConT t `AppT` ( ConT ''Route `AppT` typ)) (clazzes True)
 #else
     did <- DataInstD [] ''Route [typ] Nothing cons <$> mapM conT (clazzes False)
